File: MercaderSG/Comercial/NotaVenta/Remito/ConsultarRemito.vb
Imports Negocios
Imports Entidades
Imports Servicios
Imports Excepciones

Public Class ConsultarRemito

    Public NroNota As String

    Private Sub ConsultarRemito_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

        Me.Text = My.Resources.ArchivoIdioma.ConsultaRemitoFrm
        Dim RemDS As New GeneralDS
        Try
            NVRemitoRN.CargarRemitoNV(RemDS, NroNota)
        Catch ex As WarningException
            MessageBox.Show(ex.Message, My.Resources.ArchivoIdioma.MsgBoxAdvertencia, MessageBoxButtons.OK, MessageBoxIcon.Warning)
            GestionNV.Activate()
            Me.Close()
        End Try

        Dim Reporte As New UltimaNVRepor
        Reporte.SetDataSource(RemDS)
        RemitoCRV.ReportSource = Reporte
    End Sub

    Private Sub ConsultarRemito_KeyDown(sender As System.Object, e As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyDown
        If e.KeyCode = Keys.Escape Then
            Me.Close()
        End If
    End Sub
End Class

File: MercaderSG/Comercial/NotaVenta/Remito/ConsultarRemito.cs
using System;
using System.Windows.Forms;
using Excepciones;
using Negocios;

namespace MercaderSG
{
    public partial class ConsultarRemito
    {
        public ConsultarRemito()
        {
            InitializeComponent();
        }

        public string NroNota;

        private void ConsultarRemito_Load(object sender, EventArgs e)
        {
            Text = My.Resources.ArchivoIdioma.ConsultaRemitoFrm;
            var RemDS = new GeneralDS();
            try
            {
                NVRemitoRN.CargarRemitoNV(RemDS, NroNota);
            }
            catch (WarningException ex)
            {
                MessageBox.Show(ex.Message, My.Resources.ArchivoIdioma.MsgBoxAdvertencia, MessageBoxButtons.OK, MessageBoxIcon.Warning);
                My.MyProject.Forms.GestionNV.Activate();
                Close();
            }

            var Reporte = new UltimaNVRepor();
            Reporte.SetDataSource(RemDS);
            RemitoCRV.ReportSource = Reporte;
        }

        private void ConsultarRemito_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
            {
                Close();
            }
        }
    }
}